### ISSUE TITLE
Does not insert single-line comment tokens when wrapping a comment.

### ISSUE THREAD

**Issue by dilijev** [stkb/vscode-rewrap#8]
* Re-wrap 0.5.3 
* VSCode Version: Code - Insiders 1.4.0-insider (e3758d2cdaae78082a08a45ae28ea1bed5f47e8c, 2016-07-29T18:41:58.915Z)
* Windows 10 Build 14393.10

## Repro Steps:

Begin with (on a single line):

`// Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor congue massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra imperdiet enim. Fusce est. Vivamus a tellus. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Proin pharetra nonummy pede. Mauris et orci. Aenean nec lorem. In porttitor. Donec laoreet nonummy augue. Suspendisse dui purus, scelerisque at, vulputate vitae, pretium mattis, nunc.`

Ctrl+K Ctrl+W

End result:

```js
// Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor
congue massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus
malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra imperdiet
enim. Fusce est. Vivamus a tellus. Pellentesque habitant morbi tristique
senectus et netus et malesuada fames ac turpis egestas. Proin pharetra nonummy
pede. Mauris et orci. Aenean nec lorem. In porttitor. Donec laoreet nonummy
augue. Suspendisse dui purus, scelerisque at, vulputate vitae, pretium mattis,
nunc.
```

Expected result should be like:

```js
// Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Maecenas porttitor
// congue massa. Fusce posuere, magna sed pulvinar ultricies, purus lectus
// malesuada libero, sit amet commodo magna eros quis urna. Nunc viverra imperdiet
// enim. Fusce est. Vivamus a tellus. Pellentesque habitant morbi tristique
// senectus et netus et malesuada fames ac turpis egestas. Proin pharetra nonummy
// pede. Mauris et orci. Aenean nec lorem. In porttitor. Donec laoreet nonummy
// augue. Suspendisse dui purus, scelerisque at, vulputate vitae, pretium mattis,
// nunc.
```

Additionally, when re-wrapping multiple lines of comments beginning with a single-line comment token, the tokens should be removed before re-wrapping and re-inserted after wrapping so that stray (useless) comment tokens don't end up in the middle of a line.

**Comment by stkb** [comment]
What's the extension of the file you're using? That's exactly how it should be working already, but it is dependent on knowing the file type first, otherwise it treats everything as plain text.

**Comment by dilijev** [comment]
I opened a blank buffer and pasted that in, chose the language mode JavaScript and expected it to work. It works correctly in a *.js file.

I would argue there is still a bug here:

Considering that the user can configure different file types for the same language mode, it seems like this plugin should detect language rather than file extension and use that to choose which algorithm to enforce for re-wrapping.

**Comment by stkb** [comment]
Yeah you’re right it is (was) a bug. I noticed it too while testing it out.

Originally the extension did work like that. It only looked at the language ID provided by vscode and not the file extension. But the problem was I wanted to also support languages not supported out-of-the-box with vscode, regardless of whether the user had an extension for that language installed or not. So I had it also check the file extension for a match.

Over time and a couple of rewrites, the file extension became the preferred way to check and the language ID kinda (though not completely) got forgotten about, and no-one noticed until now :)

I've put the missing IDs back in, and it falls back to a file-extension check if the ID isn't recognised

Unless you're desperate to see it fixed right away I'd like to do a couple of other things too before publishing a new version.

**Comment by dilijev** [comment]
@stkb - no problem. I can deal with saving my files first before formatting :)

**Comment by stkb** [comment]
Fixed in v0.6.0 (comment if there's still a problem)
